src/analyzer.ts

```diff
@@ -242,30 +245,35 @@ export class Analyzer {
       // the Promise is cached before anything else happens.
       await Promise.resolve();
       const document = await this._parse(resolvedUrl, contents);
-      return this._scanDocument(document);
+      return this._scanDocument(document, false);
     })();
     this._scannedDocumentPromises.set(resolvedUrl, promise);
-    return promise;
+    const scannedDocument = await promise;
+    await this._scanDependencies(scannedDocument);
+    return scannedDocument;
   }
 
   /**
    * Parses and scans a document from source.
    */
-  private async _scanSource(
+  private async _scanInlineSource(
       type: string, contents: string, url: string,
       inlineInfo?: InlineDocInfo<any>,
       attachedComment?: string): Promise<ScannedDocument> {
     const resolvedUrl = this._resolveUrl(url);
-    const document =
+    const parsedDoc =
         this._parseContents(type, contents, resolvedUrl, inlineInfo);
-    return await this._scanDocument(document, attachedComment);
+    const scannedDoc =
+        await this._scanDocument(parsedDoc, true, attachedComment);
+    await this._scanDependencies(scannedDoc);
+    return scannedDoc;
   }
 
   /**
    * Scans a parsed Document object.
    */
   private async _scanDocument(
-      document: ParsedDocument<any, any>,
+      document: ParsedDocument<any, any>, isInline: boolean,
```

Comment:
+1 to moving this property to ParsedDocument, that's way more elegant.

When you'd want to call _scanInlineDocument vs _scan vs _scanDocument is fairly clear based on their now I think, PTAL